Problem analysis (#22)
chhoumann/p8: report/src/introduction/agile_approach.tex

\section{Implementing Agile Project Management with Scrum}
Developing a software product for a business can be a complex process.

> Incremental agile approaches have become popular to address this complexity, as it allows for fast adaptation to changes in user needs. \cite{sommervilleSoftwareEngineering2016}

Reviewer: Writing `it` is wrong as the sentence refers to multiple agile approaches.
Narrowing down to user needs is also, in my view, wrong. You learn about the problem as you explore solutions. That doesn't just include users changing needs.

Author: I think that saying it allows for faster development is a bit of a stretch though. When I read that I think 'okay, so why is this not something everyone does?'
Maybe by saying 'these methods allows for fast adaptation to rapidly changing user needs and therefore faster development of software products where this is the case.'?

> In incremental-agile approaches, system development is considered a series of increments, each adding functionality to the system.

Reviewer: Why add the dash now and not elsewhere?
Don't think this change is better. The previous describes the approach better imo.

Author: Indeed. Undone.

> In incremental-agile approaches, system development is considered a series of increments, each adding functionality to the system.

Author: ```suggestion
In incremental agile approaches, system development is considered a series of increments, each adding functionality to the system.
```

> In incremental-agile approaches, system development is considered a series of increments, each adding functionality to the system.

Author: ```suggestion
In incremental agile approaches, system development is considered a series of increments, each adding functionality to the system.
```

Between these increments, one gathers feedback from stakeholders to prioritize functionality for the next increment.
Therefore, each increment adds value to the user.

> By continuously getting feedback and working with these increments, the development process becomes cheaper and faster as less time is spent adapting to changes.\cite{sommervilleSoftwareEngineering2016}

Reviewer: Ambivalent on this one. It's still correct, and I like the brevity. However, this _seems_ to indicate that adapting to changes is what we wish to avoid... and that's not correct.

This section will discuss how an agile approach has been utilized in this project. 

> \subsection{Scrum Framework}

Reviewer: ```suggestion
\subsection*{Scrum Framework}
```

Reviewer: Or the ToC looks like this:
![image](https://user-images.githubusercontent.com/29108628/226850103-6014f439-f2dd-4970-bfe1-ac54a2429039.png)

Agile is a set of values and does not present practical activities which a process model would consist of{sutherlandScrumArtDoing2014}.

> Therefore, the team has decided on the agile framework Scrum\cite{scrumdotorg}. Scrum chosen as it puts the values to practice and due to prior experience with using the framework amongst team members. \todo{What does it mean to put the values to practice}

Reviewer: `Scrum chosen` - looks like a word is missing.

> Therefore, the team has decided on the agile framework Scrum\cite{scrumdotorg}. Scrum chosen as it puts the values to practice and due to prior experience with using the framework amongst team members. \todo{What does it mean to put the values to practice}

Author: ```suggestion
Therefore, the team has decided on the agile framework Scrum\cite{scrumdotorg}. Scrum is chosen as it puts the values to practice and due to prior experience with using the framework amongst team members. \todo{What does it mean to put the values to practice}
```

> Scrum introduces a number of practices and roles, which is described in the following sections. 

Reviewer: `is` -> `are`?
Is the change in tense intentional? Not used elsewhere.

> Scrum introduces a number of practices and roles, which is described in the following sections. 

Author: ```suggestion
Scrum introduces a number of practices and roles, which are described in the following sections. 
```

\subsubsection*{Roles and Responsibilities}

> A Scrum team consists of three roles: the Scrum Master, a Product Owner, and developers.

Reviewer: Not sure leading with a colon is great. In general, using it to list things in this kind of report makes the writing seem lazier. <- IMO

As the stakeholders provide feedback, items in the backlog should be prioritized accordingly.
All these items contribute to achieving the \textbf{product goal}.
The backlog items are \textbf{user stories}.

User stories represent functionality the stakeholder would like to see in the system.
For this reason, they must include \textit{who} it is for, \textit{what} should be done, and \textit{why} they want it.
This is captured in the general "\texttt{As a X, I want Y, such that Z.}" format for user stories.
A \textbf{sprint} is a short period of time, during which the team works on a subset of the product backlog, captured in a \textbf{sprint backlog}.
This defines the \textit{what} of the sprint.
The items in the sprint backlog are chosen based on a \textbf{sprint goal}, defining the \textit{why} of the sprint.

> The sprint has an actionable plan for delivering the increment, representing the \textit{how}. \todo{what is how? It is not mentioned anywhere else.}

Reviewer: `What`, `how`, and `why` are referenced throughout the three previous sentences to denote the functions that the aspects presented have. To say that a sprint backlog is the `what` means that it denotes what should be done. Similarly, saying that the sprint goal denotes the `why` is to say that the spring goal denotes why the sprint is being undertaken.
And finally, `how` would then be represented by the actionable plan to deliver the increment.
That is to say, we know `what` we are doing, we know `why` we are doing it, and we know `how` to do it.

> The sprint has an actionable plan for delivering the increment, representing the \textit{how}. \todo{what is how? It is not mentioned anywhere else.}

Author: ```suggestion
To establish how the user story gets completed, an actionable plan for delivering the increment is made. This plan is produced during the start of each sprint in a process called \textbf{sprint planning}.
```

> At the beginning of teach sprint, the Scrum team performs \textbf{sprint planning}.

Author: If other suggestion is okay this needs to be removed
```suggestion
```

Daily scrums are meetings where progress towards the sprint goal is discussed.
Each increment will be inspected during \textbf{sprint review}.
During such a review, progress is examined, the team and its stakeholders discuss any changes, and then figure out what to do next.

Finally, the team conducts a \textbf{sprint retrospective}, where the team examines how their last sprint went, and how they may improve.
This is part of the quest for continuous improvement.\cite{sutherlandScrumArtDoing2014}

> In summary, Scrum involves repeatedly taking backlog items and working on them during a sprint.

Reviewer: I think it makes sense to remove the section, but I really like the aesthetics of having it. It also, to me, improves the reading process by separating the summary & DoDs into their own section. 
`Artifacts and Tools` becomes very long, and its scope gets a little large.

Thoughts?

Author: ```suggestion
\subsubsection*{Scrum Process}
In summary, Scrum involves repeatedly taking backlog items and working on them during a sprint.
```
